From c49bbaeb7f6eaeb21f09fb390ac9aea12135a492 Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Wed, 25 Jul 2018 23:30:03 -0700
Subject: [PATCH 1598/1600] ASoC: SOF: do nothing if dsp suspend/resume
 callbacks are not set

Suspend/resume callbacks are currently set only for APL.
This patch will prevent errors on other platforms for which the
callbacks are not implemented as yet.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/pm.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/sof/pm.c b/sound/soc/sof/pm.c
index 063d2c9..75c1a71 100644
--- a/sound/soc/sof/pm.c
+++ b/sound/soc/sof/pm.c
@@ -210,6 +210,10 @@ static int sof_resume(struct device *dev)
 	struct snd_sof_dev *sdev = dev_get_drvdata(&priv->pdev_pcm->dev);
 	int ret = 0;
 
+	/* do nothing if dsp resume callback is not set */
+	if (!sdev->ops->resume)
+		return 0;
+
 	/* power up DSP */
 	ret = snd_sof_dsp_resume(sdev);
 	if (ret < 0) {
@@ -270,6 +274,10 @@ static int sof_suspend(struct device *dev)
 	struct snd_sof_dev *sdev = dev_get_drvdata(&priv->pdev_pcm->dev);
 	int ret = 0;
 
+	/* do nothing if dsp suspend callback is not set */
+	if (!sdev->ops->suspend)
+		return 0;
+
 	/*
 	 * Suspend running pcm streams.
 	 * They will be restarted by ALSA resume trigger call.
-- 
2.7.4

